cmd/agent/run.go

```diff
@@ -261,40 +261,70 @@ func watchExitErrors(ctx context.Context, log *logrus.Entry, exitCh chan error,
 }
 
 func runPProf(cfg config.Config, log *logrus.Entry, exitCh chan error) (closeFunc func()) {
+	log.Infof("starting pprof server on port: %d", cfg.PprofPort)
 	addr := portToServerAddr(cfg.PprofPort)
+	checkPort(addr, "pprof", log)
 	pprofSrv := &http.Server{Addr: addr, Handler: http.DefaultServeMux}
 	closeFn := func() {
+		log.Infof("closing pprof server")
```

Comment:
nit: personally, I find such related logs like "... starting", "... closing" to have identical prefix. It does sometimes result in less natural language but it helps filtering the logs.

This is such a nit that it's absolutely up to you how you like it.

Reply:
I'd prefer to keep them in natural language as they should be clear for support engineers and customers.